pkg/cache/tas_flavor_snapshot.go

```diff
@@ -344,21 +400,27 @@ func (s *TASFlavorSnapshot) buildTopologyAssignmentForLevels(domains []*domain,
 			Values: domain.levelValues[levelIdx:],
 			Count:  domain.state,
 		}
+		usage := make(resources.Requests)
+		for k, v := range singlePodRequest {
+			usage[k] = v * int64(domain.state)
+		}
+		s.addUsage(domain.id, usage)
+		s.affinityDomains = append(s.affinityDomains, domain)
 	}
 	return assignment
 }
 
-func (s *TASFlavorSnapshot) buildAssignment(domains []*domain) *kueue.TopologyAssignment {
+func (s *TASFlavorSnapshot) buildAssignment(domains []*domain, singlePodRequest resources.Requests) *kueue.TopologyAssignment {
 	// lex sort domains by their levelValues instead of IDs, as leaves' IDs can only contain the hostname
-	slices.SortFunc(domains, func(a, b *domain) int {
```

Comment:
I don't think we can drop it - when the final assignment is being constructed it is important to return the domains the tree order (lexicographical), so that the TopologyUngater correctly does rank-based ordering.